### PR TITLE
[model:stepstoreproduce] Train on defect bugs only

### LINKED ISSUE #792: Restrict the training set of the StepsToReproduce model only to defects
Given that STRs don't apply to enhancement or task.

### PR DESCRIPTION
**Pull request by naoya2000**
Fixes #792 
I am not entirely sure if my solution works properly, but I think I got the desired result because now the `get_labels` will only target those bugs whose `type` is `defect`. Also, I've used `pre-commit` to ensure that there are no style errors.

### CODE REVIEW

File: bugbug/models/stepstoreproduce.py [hunk 1]
```diff
@@ -76,18 +76,19 @@ def get_labels(self):
         classes = {}
 
         for bug_data in bugzilla.get_bugs():
-            if "cf_has_str" in bug_data:
-                if bug_data["cf_has_str"] == "no":
+            if bug_data["type"] == "defect":
```
[the remark this answers is not in the document]
Comment (naoya2000) [reply bugbug/models/stepstoreproduce.py]: Please see my explanation [here](https://github.com/naoya2000/bugbug/pull/1#issue-2046974731); hopefully I got the logic correct.

**Review by suhaibmujahid — APPROVED** [review]
Thank you, @naoya2000! LGTM!

Could you please edit the PR title to something more descriptive (e.g., "[model:stepstoreproduce] Train on defect bugs only")?

Also, could you show the model accuracy results for before and after this PR?

**Comment by naoya2000** [comment]
Here are the model accuracy results for the post-PR code. I am going to try to figure out how to restore the pre-PR code so that I can get the model accuracy results.

`
(base) naoyaokamoto@Naoyas-MacBook-Air-393 bugbug % python3 -m scripts.trainer stepstoreproduce
2023-12-29 21:03:51,287:INFO:numexpr.utils:NumExpr defaulting to 8 threads.
2023-12-29 21:03:59,004:INFO:bugbug.db:Downloading https://community-tc.services.mozilla.com/api/index/v1/task/project.bugbug.data_bugs.latest/artifacts/public/bugs.json.zst to data/bugs.json.zst
2023-12-29 21:03:59,964:INFO:__main__:Training *stepstoreproduce* model
2023-12-29 21:04:06,632:INFO:bugbug.models.stepstoreproduce:26 bugs have no steps to reproduce
2023-12-29 21:04:06,632:INFO:bugbug.models.stepstoreproduce:4462 bugs have steps to reproduce
2023-12-29 21:04:15,519:INFO:bugbug.model:X: (4488, 4), y: (4488,)
2023-12-29 21:04:20,346:INFO:bugbug.model:Cross Validation scores:
2023-12-29 21:04:20,346:INFO:bugbug.model:Accuracy: f0.6570769995215135 (+/- 0.11802274804151212)
2023-12-29 21:04:20,347:INFO:bugbug.model:Precision: f0.9975321528424977 (+/- 0.004129273673608999)
2023-12-29 21:04:20,347:INFO:bugbug.model:Recall: f0.6567870485678705 (+/- 0.12137055557161062)
2023-12-29 21:04:20,347:INFO:bugbug.model:X_train: (4039, 4), y_train: (4039,)
2023-12-29 21:04:20,347:INFO:bugbug.model:X_test: (449, 4), y_test: (449,)
2023-12-29 21:04:21,355:INFO:bugbug.model:Number of features: 73673
2023-12-29 21:04:21,355:INFO:bugbug.model:Model trained
[21:04:21] WARNING: /Users/runner/work/xgboost/xgboost/src/c_api/c_api.cc:1240: Saving into deprecated binary model format, please consider using `json` or `ubj`. Model format will default to JSON in XGBoost 2.2 if not specified.
2023-12-29 21:04:31,056:INFO:bugbug.model:Top 13 features:
+-----------+---------------------------+-----------------------------+
|   classes |   Comments contain 'been' |   Comments contain 'please' |
+===========+===========================+=============================+
|         0 |                  -0.15506 |                  -0.0987629 |
+-----------+---------------------------+-----------------------------+
|         1 |                   0.15506 |                   0.0987629 |
+-----------+---------------------------+-----------------------------+

+-----------------------------+---------------------------+----------------------------+
|   Comments contain 'report' |   Comments contain 'this' |   Comments contain 'think' |
+=============================+===========================+============================+
|                  -0.0865343 |                -0.0788646 |                 -0.0675851 |
+-----------------------------+---------------------------+----------------------------+
|                   0.0865343 |                 0.0788646 |                  0.0675851 |
+-----------------------------+---------------------------+----------------------------+

+----------------------------+----------------------------+---------------------------+
|   Comments contain 'range' |   Comments contain 'other' |   Comments contain 'like' |
+============================+============================+===========================+
|                 -0.0521299 |                 -0.0461584 |                -0.0434914 |
+----------------------------+----------------------------+---------------------------+
|                  0.0521299 |                  0.0461584 |                 0.0434914 |
+----------------------------+----------------------------+---------------------------+

+----------------------------+------------------------------+-----------------------------+
|   Comments contain 'since' |   Comments contain 'mozilla' |   Comments contain 'result' |
+============================+==============================+=============================+
|                  0.0422717 |                   -0.0299724 |                  -0.0295192 |
+----------------------------+------------------------------+-----------------------------+
|                 -0.0422717 |                    0.0299724 |                   0.0295192 |
+----------------------------+------------------------------+-----------------------------+

+---------------------------+-------------------------+
|   Comments contain 'that' |   Comments contain 'to' |
+===========================+=========================+
|                 0.0245359 |               0.0235412 |
+---------------------------+-------------------------+
|                -0.0245359 |              -0.0235412 |
+---------------------------+-------------------------+

2023-12-29 21:04:31,059:INFO:bugbug.model:Training Set scores:
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.65      1.00      0.79      0.81      0.63      4015
          0       0.02      1.00      0.65      0.03      0.81      0.68        24

avg / total       0.99      0.65      1.00      0.79      0.81      0.63      4039

2023-12-29 21:04:31,826:INFO:bugbug.model:Test Set scores:
2023-12-29 21:04:31,920:INFO:bugbug.model:No confidence threshold - 449 classified
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.64      1.00      0.78      0.80      0.62       447
          0       0.01      1.00      0.64      0.02      0.80      0.66         2

avg / total       1.00      0.64      1.00      0.78      0.80      0.62       449

╒════════════╤═════════════════╤═════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │
╞════════════╪═════════════════╪═════════════════╡
│ 1 (Actual) │             286 │             161 │
├────────────┼─────────────────┼─────────────────┤
│ 0 (Actual) │               0 │               2 │
╘════════════╧═════════════════╧═════════════════╛

2023-12-29 21:04:32,017:INFO:bugbug.model:
Confidence threshold > 0.1 - 449 classified
Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.85      1.00      0.92      0.92      0.83       447
                 0       0.03      1.00      0.85      0.06      0.92      0.86         2
__NOT_CLASSIFIED__       0.00      0.00      1.00      0.00      0.00      0.00         0

       avg / total       1.00      0.85      1.00      0.91      0.92      0.84       449

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             379 │              68 │                0 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               0 │               2 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2023-12-29 21:04:32,116:INFO:bugbug.model:
Confidence threshold > 0.2 - 449 classified
Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.76      1.00      0.87      0.87      0.74       447
                 0       0.02      1.00      0.76      0.04      0.87      0.78         2
__NOT_CLASSIFIED__       0.00      0.00      1.00      0.00      0.00      0.00         0

       avg / total       1.00      0.76      1.00      0.86      0.87      0.74       449

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             341 │             106 │                0 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               0 │               2 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2023-12-29 21:04:32,213:INFO:bugbug.model:
Confidence threshold > 0.3 - 449 classified
Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.70      1.00      0.83      0.84      0.68       447
                 0       0.01      1.00      0.70      0.03      0.84      0.73         2
__NOT_CLASSIFIED__       0.00      0.00      1.00      0.00      0.00      0.00         0

       avg / total       1.00      0.71      1.00      0.82      0.84      0.68       449

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             315 │             132 │                0 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               0 │               2 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2023-12-29 21:04:32,310:INFO:bugbug.model:
Confidence threshold > 0.4 - 449 classified
Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.67      1.00      0.80      0.82      0.65       447
                 0       0.01      1.00      0.67      0.03      0.82      0.69         2
__NOT_CLASSIFIED__       0.00      0.00      1.00      0.00      0.00      0.00         0

       avg / total       1.00      0.67      1.00      0.80      0.82      0.65       449

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             299 │             148 │                0 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               0 │               2 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2023-12-29 21:04:32,408:INFO:bugbug.model:
Confidence threshold > 0.6 - 416 classified
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.60      1.00      0.75      0.77      0.57       447
                 0       0.01      1.00      0.67      0.03      0.82      0.69         2
__NOT_CLASSIFIED__       0.00      0.00      0.93      0.00      0.00      0.00         0

       avg / total       1.00      0.60      1.00      0.74      0.77      0.57       449

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             266 │             148 │               33 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               0 │               2 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2023-12-29 21:04:32,504:INFO:bugbug.model:
Confidence threshold > 0.7 - 374 classified
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.54      1.00      0.70      0.73      0.51       447
                 0       0.01      1.00      0.70      0.03      0.84      0.73         2
__NOT_CLASSIFIED__       0.00      0.00      0.83      0.00      0.00      0.00         0

       avg / total       1.00      0.54      1.00      0.70      0.73      0.51       449

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             240 │             132 │               75 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               0 │               2 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2023-12-29 21:04:32,601:INFO:bugbug.model:
Confidence threshold > 0.8 - 315 classified
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.46      1.00      0.63      0.68      0.44       447
                 0       0.02      1.00      0.76      0.04      0.87      0.78         2
__NOT_CLASSIFIED__       0.00      0.00      0.70      0.00      0.00      0.00         0

       avg / total       1.00      0.47      1.00      0.63      0.68      0.44       449

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             207 │             106 │              134 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               0 │               2 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2023-12-29 21:04:32,701:INFO:bugbug.model:
Confidence threshold > 0.9 - 221 classified
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.34      1.00      0.51      0.58      0.32       447
                 0       0.03      1.00      0.85      0.06      0.92      0.86         2
__NOT_CLASSIFIED__       0.00      0.00      0.49      0.00      0.00      0.00         0

       avg / total       1.00      0.34      1.00      0.50      0.58      0.32       449

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             151 │              68 │              228 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               0 │               2 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2023-12-29 21:04:32,741:INFO:__main__:Training done
tar: Couldn't open zstdmt: No such file or directory
2023-12-29 21:04:32,758:WARNING:bugbug.utils:Command '['tar', '-I', 'zstdmt', '-cf', 'stepstoreproducemodel.tar.zst', 'stepstoreproducemodel']' returned non-zero exit status 1.. Falling back to zstandard API, which could be slower.
2023-12-29 21:04:32,776:INFO:__main__:Model compressed
`

**Comment by suhaibmujahid** [comment]
Thank you, @naoya2000! 

>  I am going to try to figure out how to restore the pre-PR code so that I can get the model accuracy results.

You can switch to the master branch and train the model from there. I did that, and the results seem similar:

```
2024-01-02 09:00:46,462:INFO:bugbug.model:Top 14 features:
+-----------+-------------------------+-------------------------+
|   classes |   Comments contain 'we' |   Comments contain 'so' |
+===========+=========================+=========================+
|         0 |               -0.127059 |              -0.0961756 |
+-----------+-------------------------+-------------------------+
|         1 |                0.127059 |               0.0961756 |
+-----------+-------------------------+-------------------------+

+-----------------------------+------------------------------+----------------------------+
|   Comments contain 'please' |   Comments contain 'comment' |   Comments contain 'could' |
+=============================+==============================+============================+
|                  -0.0859485 |                   -0.0740519 |                 -0.0608341 |
+-----------------------------+------------------------------+----------------------------+
|                   0.0859485 |                    0.0740519 |                  0.0608341 |
+-----------------------------+------------------------------+----------------------------+

+------------------------------+-----------------------------+-----------------------------+
|   Comments contain '__url__' |   Comments contain 'thread' |   Comments contain 'actual' |
+==============================+=============================+=============================+
|                   -0.0537922 |                  -0.0517098 |                  -0.0494718 |
+------------------------------+-----------------------------+-----------------------------+
|                    0.0537922 |                   0.0517098 |                   0.0494718 |
+------------------------------+-----------------------------+-----------------------------+

+---------------------------+--------------------------+------------------------------+
|   Comments contain 'make' |   Comments contain 'you' |   Comments contain 'created' |
+===========================+==========================+==============================+
|                -0.0414499 |                0.0397537 |                   -0.0386979 |
+---------------------------+--------------------------+------------------------------+
|                 0.0414499 |               -0.0397537 |                    0.0386979 |
+---------------------------+--------------------------+------------------------------+

+----------------------------+------------------------------+----------------------------+
|   Comments contain 'issue' |   Comments contain 'mozilla' |   Comments contain 'other' |
+============================+==============================+============================+
|                  0.0245583 |                    -0.023821 |                 -0.0223592 |
+----------------------------+------------------------------+----------------------------+
|                 -0.0245583 |                     0.023821 |                  0.0223592 |
+----------------------------+------------------------------+----------------------------+

2024-01-02 09:00:46,465:INFO:bugbug.model:Training Set scores:
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.72      1.00      0.84      0.85      0.70      4257
          0       0.02      1.00      0.72      0.04      0.85      0.74        26

avg / total       0.99      0.72      1.00      0.83      0.85      0.70      4283

2024-01-02 09:00:47,493:INFO:bugbug.model:Test Set scores:
2024-01-02 09:00:47,633:INFO:bugbug.model:No confidence threshold - 476 classified
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.71      0.00      0.83      0.00      0.00       475
          0       0.00      0.00      0.71      0.00      0.00      0.00         1

avg / total       0.99      0.71      0.00      0.83      0.00      0.00       476

╒════════════╤═════════════════╤═════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │
╞════════════╪═════════════════╪═════════════════╡
│ 1 (Actual) │             339 │             136 │
├────────────┼─────────────────┼─────────────────┤
│ 0 (Actual) │               1 │               0 │
╘════════════╧═════════════════╧═════════════════╛

2024-01-02 09:00:47,782:INFO:bugbug.model:
Confidence threshold > 0.1 - 476 classified
Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.92      0.00      0.96      0.00      0.00       475
                 0       0.00      0.00      0.92      0.00      0.00      0.00         1
__NOT_CLASSIFIED__       0.00      0.00      1.00      0.00      0.00      0.00         0

       avg / total       1.00      0.92      0.00      0.96      0.00      0.00       476

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             437 │              38 │                0 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               1 │               0 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2024-01-02 09:00:47,926:INFO:bugbug.model:
Confidence threshold > 0.2 - 476 classified
Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.85      0.00      0.92      0.00      0.00       475
                 0       0.00      0.00      0.85      0.00      0.00      0.00         1
__NOT_CLASSIFIED__       0.00      0.00      1.00      0.00      0.00      0.00         0

       avg / total       1.00      0.84      0.00      0.91      0.00      0.00       476

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             402 │              73 │                0 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               1 │               0 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2024-01-02 09:00:48,072:INFO:bugbug.model:
Confidence threshold > 0.3 - 476 classified
Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.79      0.00      0.88      0.00      0.00       475
                 0       0.00      0.00      0.79      0.00      0.00      0.00         1
__NOT_CLASSIFIED__       0.00      0.00      1.00      0.00      0.00      0.00         0

       avg / total       1.00      0.78      0.00      0.88      0.00      0.00       476

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             373 │             102 │                0 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               1 │               0 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2024-01-02 09:00:48,217:INFO:bugbug.model:
Confidence threshold > 0.4 - 476 classified
Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.76      0.00      0.86      0.00      0.00       475
                 0       0.00      0.00      0.76      0.00      0.00      0.00         1
__NOT_CLASSIFIED__       0.00      0.00      1.00      0.00      0.00      0.00         0

       avg / total       1.00      0.75      0.00      0.86      0.00      0.00       476

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             359 │             116 │                0 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               1 │               0 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2024-01-02 09:00:48,364:INFO:bugbug.model:
Confidence threshold > 0.6 - 434 classified
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.67      0.00      0.80      0.00      0.00       475
                 0       0.00      0.00      0.76      0.00      0.00      0.00         1
__NOT_CLASSIFIED__       0.00      0.00      0.91      0.00      0.00      0.00         0

       avg / total       0.99      0.67      0.00      0.80      0.00      0.00       476

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             317 │             116 │               42 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               1 │               0 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2024-01-02 09:00:48,510:INFO:bugbug.model:
Confidence threshold > 0.7 - 397 classified
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.62      0.00      0.76      0.00      0.00       475
                 0       0.00      0.00      0.79      0.00      0.00      0.00         1
__NOT_CLASSIFIED__       0.00      0.00      0.83      0.00      0.00      0.00         0

       avg / total       0.99      0.62      0.00      0.76      0.00      0.00       476

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             294 │             102 │               79 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               1 │               0 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2024-01-02 09:00:48,656:INFO:bugbug.model:
Confidence threshold > 0.8 - 318 classified
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.51      0.00      0.68      0.00      0.00       475
                 0       0.00      0.00      0.85      0.00      0.00      0.00         1
__NOT_CLASSIFIED__       0.00      0.00      0.67      0.00      0.00      0.00         0

       avg / total       0.99      0.51      0.00      0.68      0.00      0.00       476

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             244 │              73 │              158 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               1 │               0 │                0 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2024-01-02 09:00:48,802:INFO:bugbug.model:
Confidence threshold > 0.9 - 223 classified
Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
Sensitivity is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.
                          pre       rec       spe        f1       geo       iba       sup

                 1       1.00      0.39      1.00      0.56      0.62      0.37       475
                 0       0.00      0.00      0.92      0.00      0.00      0.00         1
__NOT_CLASSIFIED__       0.00      0.00      0.47      0.00      0.00      0.00         0

       avg / total       1.00      0.39      1.00      0.56      0.62      0.36       476

╒════════════╤═════════════════╤═════════════════╤══════════════════╕
│            │   1 (Predicted) │   0 (Predicted) │   Not classified │
╞════════════╪═════════════════╪═════════════════╪══════════════════╡
│ 1 (Actual) │             185 │              38 │              252 │
├────────────┼─────────────────┼─────────────────┼──────────────────┤
│ 0 (Actual) │               0 │               0 │                1 │
╘════════════╧═════════════════╧═════════════════╧══════════════════╛

2024-01-02 09:00:48,883:INFO:__main__:Training done
```

**Pull request merged**